### PR TITLE
Configure eth.staging fleet

### PR DESCRIPTION
**Pull request by adambabik**
Provide a way to run `statusd` with `eth.staging` fleet.

### CODE REVIEW

File: params/config.go [hunk 1]
```diff
@@ -604,15 +613,24 @@ func (c *NodeConfig) updateClusterConfig() error {
 			return fmt.Errorf("failed to unmarshal cluster configuration file: %s", err)
 		}
 	} else {
-		clusters = defaultClusters
+		switch c.ClusterConfig.Fleet {
+		case fleetStaging:
+			clusters = stagingFleet.Cluster
+		default:
+			clusters = defaultFleet.Cluster
+		}
```
Comment (jakubgs) [inline params/config.go]: Shouldn't this be written in such a way that we can add a fleet to the config without having to mess about with the code?

**Review by jakubgs — CHANGES_REQUESTED** [review]

File: params/config.go (hunk 1 above)
Comment (adambabik) [reply params/config.go]: Yes, good point. I have just changed that.

File: cmd/statusd/main.go [hunk 2]
```diff
@@ -92,7 +93,7 @@ var (
 // All general log messages in this package should be routed through this logger.
 var logger = log.New("package", "status-go/cmd/statusd")
 
-func main() {
+func init() {
```
Comment (divan) [inline cmd/statusd/main.go:98]: I would keep flag parsing in `main()`. Imagine, `init` function will be moved into another file or just to the bottom of file, and anyone looking at the `func main() {` will loose the fact that the flag parsing happens.
Also, let's keep PRs as small as possible – if there is a sense to refactor main/init code, let's do it in a separate PR, it has nothing to do with fleet configuration.

File: cmd/statusd/whispercfg.go [hunk 3]
```diff
@@ -12,11 +12,13 @@ import (
 // whisperConfig creates node configuration object from flags
 func whisperConfig(nodeConfig *params.NodeConfig) (*params.NodeConfig, error) {
 	whisperConfig := nodeConfig.WhisperConfig
+
 	whisperConfig.Enabled = true
 	whisperConfig.EnableMailServer = *enableMailServer
 	whisperConfig.LightClient = *lightClient
 	whisperConfig.MinimumPoW = *minPow
 	whisperConfig.TTL = *ttl
+	whisperConfig.EnableNTPSync = *ntpSyncEnabled
```
Comment (divan) [inline cmd/statusd/whispercfg.go:21]: the same – it seems like a totally separate problem fixed here, not related to fleet configuration

File: params/config.go [hunk 4]
```diff
@@ -344,7 +347,7 @@ type NodeConfig struct {
 }
 
 // NewNodeConfig creates new node configuration object
-func NewNodeConfig(dataDir string, clstrCfgFile string, networkID uint64) (*NodeConfig, error) {
+func NewNodeConfig(dataDir, clstrCfgFile, fleet string, networkID uint64) (*NodeConfig, error) {
```
Comment (divan) [inline params/config.go:356]: With 4 parameters, semantics of `NewNodeConfig()` becomes less readable. Maybe worth rethinking how we use this node config stuff (in separate PR)? I.e. `params.NewNodeConfig("", "", "", 0)` is not the extra clear :)

File: cmd/statusd/main.go (hunk 2 above)
Comment (adambabik) [reply cmd/statusd/main.go:98]: I am not sure why `init()` would be moved anywhere else. It is executed before `main()` so logically it should be before it.

I needed to make some changes because the logger in `NodeConfig` was improperly initialized before we initialize the root logger object.

File: params/config.go (hunk 4 above)
Comment (adambabik) [reply params/config.go:356]: I agree. I think the whole `params` package needs some rethinking.

File: cmd/statusd/main.go (hunk 2 above)
Comment (divan) [reply cmd/statusd/main.go:98]: Hmm, okay, if there is a valid reason than it's fine. Generally, init should be avoided unless really needed, because init() is one of those rare "magical" things that Go has, and they can easily fool/conceal  important aspects of the program. Anyway,

**Review by gravityblast — APPROVED** [review]

**Review by divan — APPROVED** [review]
LGTM
